packages/block-editor/src/components/rich-text/index.js

```diff
@@ -159,6 +159,18 @@ function RichTextWrapper(
 	// retreived from the store on merge.
 	// To do: fix this somehow.
 	const { selectionStart, selectionEnd, isSelected } = useSelect( selector );
+	const canInsertEmbed = useSelect(
+		( select ) => {
+			const { canInsertBlockType, getBlockRootClientId } =
+				select( blockEditorStore );
+
+			return canInsertBlockType(
+				'core/embed',
+				getBlockRootClientId( clientId )
+			);
+		},
+		[ clientId ]
+	);
```

Comment:
I decided to use a separate selector since, like the first selector, this doesn't need to run on every render.